modules/pairIdSystem.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * This module adds PAIR Id to the User ID module
+ * The {@link module:modules/userId} module is required
+ * @module modules/pairIdSystem
+ * @requires module:modules/userId
+ */
+
+import { submodule } from '../src/hook.js';
+import {getStorageManager} from '../src/storageManager.js'
+import { logError } from '../src/utils.js';
+
+const MODULE_NAME = 'pairId';
+const PAIR_ID_KEY = 'pairId';
+const DEFAULT_LIVERAMP_PAIR_ID_KEY = '_lr_pairId';
+
+export const storage = getStorageManager()
+
+function pairIdFromLocalStorage(key) {
+  return storage.localStorageIsEnabled ? storage.getDataFromLocalStorage(key) : null;
+}
+
+function pairIdFromCookie(key) {
+  return storage.cookiesAreEnabled ? storage.getCookie(key) : null;
+}
+
+/** @type {Submodule} */
+export const pairIdSubmodule = {
+  /**
+  * used to link submodule with config
+  * @type {string}
+  */
+  name: MODULE_NAME,
+  /**
+   * decode the stored id value for passing to bid requests
+   * @function
+   * @param { string | undefined } value
+   * @returns {{pairId:string} | undefined }
+   */
+  decode(value) {
+    return value && Array.isArray(value) ? {'pairId': value} : undefined
+  },
+  /**
+  * performs action to obtain id and return a value in the callback's response argument
+  * @function
+  * @returns {id: string | undefined }
+  */
+  getId(config) {
+    const pairIdsString = pairIdFromLocalStorage(PAIR_ID_KEY) || pairIdFromCookie(PAIR_ID_KEY)
+    let ids = []
+    if (pairIdsString && typeof pairIdsString == 'string') {
+      try {
+        ids = ids.concat(JSON.parse(atob(pairIdsString)))
+      } catch (error) {
+        logError(error)
+      }
+    }
+
+    const configParams = (config && config.params) || {};
+    if (configParams && configParams.liveramp) {
+      let LRStorageLocation = configParams.liveramp.storageKey || DEFAULT_LIVERAMP_PAIR_ID_KEY
+      const liverampValue = pairIdFromLocalStorage(LRStorageLocation) || pairIdFromCookie(LRStorageLocation)
+      try {
+        const obj = JSON.parse(atob(liverampValue));
+        ids = ids.concat(obj.envelope);
+      } catch (error) {
+        logError(error)
```

Comment:
if some other code does not set the pair id or the liveramp id in storage or cookie then this module logs an error every single time. Which seems like overkill.

Same with the check below this, if no ID is found it throws a pretty nasty error:

<img width="569" alt="image" src="https://github.com/prebid/Prebid.js/assets/13972794/3d518b44-64ae-4b2d-9407-08ce3ee560a9">


Perhaps these can be changed to WARN or INFO messages instead?

Example of error where pair id is not found:

<img width="429" alt="image" src="https://github.com/prebid/Prebid.js/assets/13972794/9b7eb15d-6ba0-4ce8-aa7c-a0676761ab23">


I think an error is a bit too strong here.